opentelemetry-auto-annotations-beta-shaded-for-instrumenting/opentelemetry-auto-annotations-beta-shaded-for-instrumenting.gradle

```diff
@@ -0,0 +1,17 @@
+plugins {
+  id "com.github.johnrengelman.shadow"
+}
+
+apply from: "$rootDir/gradle/java.gradle"
+
+dependencies {
+  implementation deps.opentelemetryAutoAnnotations
+}
+
+// OpenTelemetry Auto Annotations shaded so that it can be used in opentelemetry-api-beta
+// instrumentation, and then its usage can be unshaded after OpenTelemetry API is shaded
```

Comment:
Just to confirm this means our unit tests can catch the issue in the future?

Reply:
unfortunately, no, this was needed for the fix. currently at least we have to rely on smoke tests to catch shading-related problems. probably we should create a whole smoke test around instrumentation OpenTelemetry API, since that instrumentation is shading-fragile. i will open issue........ (a couple min later) oh ya, i have an issue already for that: #298 😄